app/src/app/uploader/page.tsx

```diff
@@ -0,0 +1,154 @@
+'use client';
+import React, {useState} from 'react';
+import Papa from 'papaparse';
+import {ErrorNotification} from '../components/ErrorNotification';
+import {GerryDBViewSelector} from '../components/sidebar/GerryDBViewSelector';
+import {useMapStore} from '@/app/store/mapStore';
+import {Assignment, createMapDocument, uploadAssignments} from '@/app/utils/api/apiHandlers';
+
+type MapLink = {
+  document_id: string;
+  name: string;
+};
+
+export default function Uploader() {
+  const [progress, setProgress] = useState<number>(0);
+  const [totalRows, setTotalRows] = useState<number>(0);
+  const [mapLinks, setMapLinks] = useState<MapLink[]>([]);
+
+  const ROWS_PER_BATCH = 20000000000;
+
+  const gTable = useMapStore(state => state.mapDocument?.gerrydb_table);
+  const upsertUserMap = useMapStore(state => state.upsertUserMap);
+
+  const handleDragOver = (event: React.DragEvent) => {
+    event.preventDefault();
+  };
+
+  const processFile = (file: File) => {
+    if (!file) {
+      throw new Error('No file selected');
+      return;
```

Comment:
H: At least some file type validation would be good -- I accidentally dropped in a parquet and only got an internal server error response.

Additionally, some simple field / geoid validation would be good -- eg. if more than two columns should we ask users? Checking if GEOIDs all start with the expected state FIPS code?